htdocs/install/repair.php

```diff
@@ -2016,14 +2019,23 @@
 						fd.fk_facture = $obj->rowid";
 				$ressql_calculs = $db->query($sql_calculs);
 				while ($obj_calcul = $db->fetch_object($ressql_calculs)) {
+					// Calcul de la somme des paiements reçus
+					$sql_paiements = "SELECT SUM(amount) as somme from ".MAIN_DB_PREFIX."paiement_facture WHERE fk_facture = $obj->rowid";
+					$montantPaiements = $db->fetch_object($db->query($sql_paiements))->somme;
+					$totHt= ($obj_calcul->total_ht ? price2num($obj_calcul->total_ht, 'MT') : 0);
+					$totTva = ($obj_calcul->total_tva ? price2num($obj_calcul->total_tva, 'MT') : 0);
+					$totLocal1 = ($obj_calcul->localtax1 ? price2num($obj_calcul->localtax1, 'MT') : 0);
+					$totLocal2 = ($obj_calcul->localtax2 ? price2num($obj_calcul->localtax2, 'MT') : 0);
+					$totTtc = $totHt + $totTva + $totLocal1 + $totLocal2;
 					$sql_maj = "
 						UPDATE ".MAIN_DB_PREFIX."facture
 						SET
-							total_ht = ".($obj_calcul->total_ht ? price2num($obj_calcul->total_ht, 'MT') : 0).",
-							total_tva = ".($obj_calcul->total_tva ? price2num($obj_calcul->total_tva, 'MT') : 0).",
-							localtax1 = ".($obj_calcul->localtax1 ? price2num($obj_calcul->localtax1, 'MT') : 0).",
-							localtax2 = ".($obj_calcul->localtax2 ? price2num($obj_calcul->localtax2, 'MT') : 0).",
-							total_ttc = ".($obj_calcul->total_ttc ? price2num($obj_calcul->total_ttc, 'MT') : 0)."
+							total_ht = $totHt,
+							total_tva = $totTva,
+							localtax1 = $totLocal1,
+							localtax2 = $totLocal2,
+							total_ttc = $totTtc,
+							fk_statut = ".($totTtc == price2num($montantPaiements, 'MT') ? 2 : 1 )."
```

Comment:
Not that situation should not happen because total should be rounded in database so total of all taxes + total_ht should match total_ttc.
Did you know how to reproduce the corruption of data ?